digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/LinearLocationDAO.scala

```diff
@@ -0,0 +1,731 @@
+        ps.setLong(3, location.orderNumber)
+        ps.setLong(4, location.linkId)
+        ps.setDouble(5, location.startMValue)
+        ps.setDouble(6, location.endMValue)
+        ps.setInt(7, location.sideCode.value)
+        location.startCalibrationPoint match {
+          case Some(value) => ps.setLong(8, value)
+          case None => ps.setNull(8, Types.BIGINT)
+        }
+        location.endCalibrationPoint match {
+          case Some(value) => ps.setLong(9, value)
+          case None => ps.setNull(9, Types.BIGINT)
+        }
+        ps.setInt(10, location.linkGeomSource.value)
+        ps.setLong(11, location.adjustedTimestamp)
+        ps.setInt(12, location.floating.value)
+        val (p1, p2) = (location.geometry.head, location.geometry.last)
+        ps.setDouble(13, p1.x)
+        ps.setDouble(14, p1.y)
+        ps.setDouble(15, location.startMValue)
+        ps.setDouble(16, p2.x)
+        ps.setDouble(17, p2.y)
+        ps.setDouble(18, location.endMValue)
+        ps.setString(19, if (createdBy == null) "-" else createdBy)
+        ps.addBatch()
+    }
+    ps.executeBatch()
+    ps.close()
+    createLinearLocations.map(_.id).toSeq
+  }
+
+  def lockLinearLocationWriting: Unit = {
+    sqlu"""LOCK TABLE linear_location IN SHARE MODE""".execute
+  }
+
+  implicit val getLinearLocation: GetResult[LinearLocation] = new GetResult[LinearLocation] {
+    def apply(r: PositionedResult) = {
+      val id = r.nextLong()
+      val roadwayNumber = r.nextLong()
+      val orderNumber = r.nextLong()
+      val linkId = r.nextLong()
+      val startMeasure = r.nextDouble()
+      val endMeasure = r.nextDouble()
+      val sideCode = r.nextInt()
+      val calStartM = r.nextLongOption()
+      val calEndM = r.nextLongOption()
+      val linkSource = r.nextInt()
+      val adjustedTimestamp = r.nextLong()
+      val floating = r.nextInt()
+      val x1 = r.nextDouble()
+      val y1 = r.nextDouble()
+      val x2 = r.nextDouble()
+      val y2 = r.nextDouble()
+      val validFrom = r.nextDateOption.map(d => formatter.parseDateTime(d.toString))
+      val validTo = r.nextDateOption.map(d => formatter.parseDateTime(d.toString))
+
+      LinearLocation(id, orderNumber, linkId, startMeasure, endMeasure, SideCode.apply(sideCode), adjustedTimestamp,
+        (calStartM, calEndM), FloatingReason.apply(floating), Seq(Point(x1, y1), Point(x2, y2)),
+        LinkGeomSource.apply(linkSource), roadwayNumber, validFrom, validTo)
+    }
+  }
+
+  def fetchLinkIdsInChunk(min: Long, max: Long): List[Long] = {
+    sql"""
+      select distinct(loc.link_id)
+      from linear_location loc where loc.link_id between $min and $max order by loc.link_id asc
+    """.as[Long].list
+  }
+
+  private def queryList(query: String): List[LinearLocation] = {
+    Q.queryNA[LinearLocation](query).list.groupBy(_.id).map {
+      case (_, list) =>
+        list.head
+    }.toList
+  }
+
+  def fetchById(id: Long): Option[LinearLocation] = {
```

Comment:
Should we update the names of the methods?
Right now we have "fetchById" and "queryById", and they do the same operation, but one for a single id and the other for a list of ids. Should we keep different names? We have more cases like this one